### PR TITLE
Add TermIdsAcquirer interface and in-memory implementation

### PR DESCRIPTION
**Pull request by AlaaSarhan**

### CODE REVIEW

File: src/PackagePrivate/InMemoryTermIdsAcquirer.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+		foreach ( $termsArray as $type => $termsOfType ) {
+			foreach ( $termsOfType as $lang => $terms ) {
+				$terms = (array)$terms;
+
+				foreach ( $terms as $term ) {
+					if ( isset( $this->terms[$type][$lang][$term] ) ) {
+						$ids[] = $this->terms[$type][$lang][$term];
+					} else {
+						$this->terms[$type][$lang][$term] = ++$this->lastId;
```
Comment (thiemowmde) [inline src/PackagePrivate/InMemoryTermIdsAcquirer.php]: This uses `$this->terms` as if it is an array, but it is never initialized. Can you please add `= []` to the property declaration?

Comment (AlaaSarhan) [reply src/PackagePrivate/InMemoryTermIdsAcquirer.php]: yes sure

File: tests/Unit/PackagePrivate/InMemoryTermIdsAcquirerTest.php [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+			'alias' => [
+				'en' => [ 'alias', 'another alias' ],
+				'de' => 'de alias'
+			]
+		] );
+
+		$this->assertEquals(
+			$ids,
+			array_unique( $ids )
+		);
```
Comment (JeroenDeDauw) [inline tests/Unit/PackagePrivate/InMemoryTermIdsAcquirerTest.php]: You can extract this into a new assertNoDuplicates method

**Comment by JeroenDeDauw** [comment]
I didn't sleep well and this weird mate drink only helped so much. So still can't review if this makes sense properly. Third day is a charm?

image

**Comment by AlaaSarhan** [comment]
I'm self-merging this one to unblock dependent work for now

**Pull request merged**